### PR TITLE
Add type annotations to `MFT.py` and `BinaryParser.py`

### PR DESCRIPTION
**Pull request by sldouglas-nist**
The one patch's log message describes its purpose.

### CODE REVIEW

File: indxparse/MFT.py [hunk 1]
```diff
@@ -1201,7 +1208,7 @@ def __init__(
 
         self.fixup(self.usa_count(), self.usa_offset())
 
-    def attributes(self):
+    def attributes(self) -> typing.Generator[Attribute, None, None]:
```
Comment (williballenthin) [inline indxparse/MFT.py:1211]: using `Iterator[Attribute]` is a bit simpler

File: indxparse/MFT.py [hunk 2]
```diff
@@ -1219,14 +1226,14 @@ def attribute(self, attr_type):
             if a.type() == attr_type:
                 return a
 
-    def is_directory(self):
+    def is_directory(self) -> int:
```
Comment (williballenthin) [inline indxparse/MFT.py:1229]: should be bool

File: indxparse/MFT.py [hunk 3]
```diff
@@ -1219,14 +1226,14 @@ def attribute(self, attr_type):
...
         return self.flags() & MFT_RECORD_FLAGS.MFT_RECORD_IS_DIRECTORY
 
-    def is_active(self):
+    def is_active(self) -> int:
```
Comment (williballenthin) [inline indxparse/MFT.py:1232]: bool

**Review by williballenthin — CHANGES_REQUESTED** [review]
some minor tweaks inline. 

thanks for the PR!

File: indxparse/MFT.py (hunk 1 above)
Comment (sldouglas-nist) [reply indxparse/MFT.py]: Hi Willi, this was addressed in a recent patch.

File: indxparse/MFT.py (hunk 2 above)
Comment (sldouglas-nist) [reply indxparse/MFT.py]: Hi Willi, these `bool` comments were addressed in a recent patch.

**Review by williballenthin — CHANGES_REQUESTED** [review]
one tweak, then ready to go

File: indxparse/MFT.py [hunk 4]
```diff
@@ -1333,7 +1340,7 @@ def _calculate_mftoffset(self):
             self.mftoffset = self.offset + relmftoffset * self.clustersize
             logging.debug("MFT offset is %s", hex(self.mftoffset))
 
-    def record_generator(self, start_at=0):
+    def record_generator(self, start_at=0) -> typing.Generator[MFTRecord, None, None]:
```
[the remark this answers is not in the document]
Comment (sldouglas-nist) [reply indxparse/MFT.py]: Hi Willi, this was addressed in a recent patch.

**Review by williballenthin — APPROVED** [review]
thank you @sldouglas-nist!

**Pull request merged**